content/lib/pencil-tracer.js

```diff
@@ -731,8 +775,37 @@
       }
     };
 
+    JavaScriptInstrumenter.prototype.isFunctionDef = function(node) {
+      var ref;
+      return (node != null ? node.type : void 0) === "FunctionDeclaration" || ((node != null ? node.type : void 0) === "VariableDeclaration" && node.declarations.length === 1 && ((ref = node.declarations[0].init) != null ? ref.type : void 0) === "FunctionExpression");
+    };
+
+    JavaScriptInstrumenter.prototype.soakify = function(name) {
+      var closeParens, expr, i, j, parts, ref, soakified;
+      soakified = "";
+      closeParens = "";
+      parts = name.split(".");
+      for (i = j = 0, ref = parts.length; 0 <= ref ? j < ref : j > ref; i = 0 <= ref ? ++j : --j) {
```

Comment:
I have no idea what this is doing. Better variable names.
It looks like you've used this pattern elsewhere. Consider writing a function that does the iteration you want (in the body of the "for", you don't change i or j, so apparently this is some fancy iteration over i and j), and then have it call a "body" function that does the rest. Remember that Javascript does closures, so the body function can naturally refer to variables in the outer scope.

Ask me for explanation in person.  ;-)